liquibase-core/src/main/java/liquibase/changelog/ChangeLogParameters.java

```diff
@@ -117,21 +118,25 @@ public void set(String key, String value, String contexts, String labels, String
     public void set(String key, String value, ContextExpression contexts, Labels labels, String databases,
                     boolean globalParam, DatabaseChangeLog changeLog) {
         /**
-         * TODO: this was a bug. Muliple created parameters have been created, but the corresponding method in
+         * TODO: this was a bug. Multiple created parameters have been created, but the corresponding method in
          * #findParameter() is only catching the first one. So here we should eliminate duplicate entries
          **/
         if (globalParam) {
-            // if it is global param ignore additional adds
+            // if it is global param remove duplicate non-global parameters
             ChangeLogParameter param = findParameter(key, null);
-            if (param == null) {
-                // okay add it
-                changeLogParameters.add(new ChangeLogParameter(key, value, contexts, labels, databases, globalParam,
-                    changeLog));
+            if (param != null && ! param.isGlobal()) {
+                changeLogParameters.remove(param);
             }
-        } else {
-            //this is a non-global param, just add it
+            // okay add it
             changeLogParameters.add(new ChangeLogParameter(key, value, contexts, labels, databases, globalParam,
```

Comment:
I worry that this breaks logic around "once a parameter is set, you can't change the value". Since it looks like we add the value to the changelogParameters list all the time now.

But, the automated tests pass and seem to be checking for that behavior pretty well, so must be OK??